src/errors/errors.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+class FetchBaseError extends Error {
+	constructor(message, type) {
+		super(message);
+		// Hide custom error implementation details from end-users
+		Error.captureStackTrace(this, this.constructor);
+
+		this.type = type;
```

Comment:
I've generally done the following to provide a read-only property in errors:

```suggestion
		Object.defineProperty(this, 'type', {
			configurable: false,
			enumerable: true,
			value: type,
			writable: false,
		});
```

Reply:
Well. Do you hate inheritance in that case, don't you?

Reply:
That is a rather rude, borderline personal attack to a friendly suggestion.

:man_shrugging: 

It's a property passed into the super's constructor and generally would not be changed after it is initially set. Making it read-only seemed, to me, to be a logical conclusion. It's also a pattern I've seen a few times in other projects for a `type` property in error classes.

Reply:
Sorry, @MitMaro I didn't mean to talk about personality,  my bad, I mean just _class inheritance_. But by making the `type` non-configurable property set in base class how we suppose to override it in a child class?